scrapy_poet/middleware.py

```diff
@@ -64,20 +64,21 @@ def process_response(self, request: Request, response: Response,
         """
         # Find out the dependencies
         callback = utils.get_callback(request, spider)
-        dependencies = {
+        external_dependencies = {
```

Comment:
There are two types of external dependencies:

- The external dependencies required to build the providers
- The external dependencies required to invoke the callback (which are provided by the providers). 

The `external_dependencies` name is ambiguous and confused me when reading the code. It is now passed as argument to some methods and was thinking about which kind of dependencies they were. 

I propose then to change the name to make this more clear. What for example `provider_dependencies`? `provider_external_dependencies` would probably be the most accurate, but too long I guess. Anyway, this is very minor.

Reply:
data point: I only understood what dependencies vs external_dependencies are after a few hours of code review, and forgot about it a few hours later :)

Reply:
Agree with you guys, just couldn't think about a better name.

Reply:
Thinking on more names: `externally_available_for_providers` and `externally_available_for_callbacks` Quite long, uhh!

`external_types_for_providers` and `external_types_for_callbacks` are shorter. 

`supported_init_types_for_providers` and `types_supplied_by_providers`

Reply:
Why not `scrapy_objects` then?

Reply:
Yeah, sounds good.